addig more details of how to define the error code and error message via claimstrasformatios  Update oauth2-error-technical-profile.md (#121333)
Azure/azure-content: articles/active-directory-b2c/oauth2-error-technical-profile.md

</ClaimsProviders> -->
```

> ## Define claimstransformations to generate custom value of error code and error message 

Reviewer: ```suggestion
## Define claims transformation to generate custom values of error code and error message 
```

> 

Reviewer: Use these steps to generate custom values of error code and error message:

1. Locate the `ClaimsTransformations` element, then add the following code inside it:

```xml
<!--
 <ClaimsTransformations> -->
<ClaimsTransformation Id="GenerateErrorCode" TransformationMethod="CreateStringClaim">
        <InputParameters>
          <InputParameter Id="value" DataType="string" Value="Error_001" />
        </InputParameters>
        <OutputClaims>
          <OutputClaim ClaimTypeReferenceId="errorCode" TransformationClaimType="createdClaim" />
        </OutputClaims>
      </ClaimsTransformation>
      <ClaimsTransformation Id="GenerateErrorMessage" TransformationMethod="CreateStringClaim">
        <InputParameters>
          <InputParameter Id="value" DataType="string" Value="Insert error description." />
        </InputParameters>
        <OutputClaims>
          <OutputClaim ClaimTypeReferenceId="errorMessage" TransformationClaimType="createdClaim" />
        </OutputClaims>
      </ClaimsTransformation>
<!--
</ClaimsTransformations> -->
```

> And you will need to add these 2 ClaimsTransformations in OutputClaimsTransformations/InputClaimsTransformations of any Technical Profiles before Oauth2 TP you defined:

Reviewer: ```suggestion
1. Add the two claims transformations in the `OutputClaimsTransformations` element of any technical profile before Oauth2 technical that you define:
```

Reviewer: @SiverShiSSS I don't understand this statement, can you elaborate. And do we add in both the `OutputClaimsTransformations` element and `InputClaimsTransformations` element or just in the `OutputClaimsTransformations` element?

```xml
          <OutputClaimsTransformations>
            <OutputClaimsTransformation ReferenceId="generateErrorCode" />
            <OutputClaimsTransformation ReferenceId="generateErrorMessage" />

>           </outputClaimsTransformations>

Reviewer: ```suggestion
          </OutputClaimsTransformations>
```

</UserJourney>
```

> *Optional: And you can add/use other proconditions to manipulate the calling of Oauth2 error TP. For example, we can set to call Oauth2 error TP if there is no email claims.

Reviewer: ```suggestion
Optionally, you can use preconditions to manipulate the Oauth2 error technical profile. For example, if there is no email claim, you can set to call Oauth2 error technical profile.
```
